dvc/checkout.py

```diff
@@ -243,6 +251,9 @@ def _checkout_dir(
         or modified
     )
 
+    if failed:
+        raise CheckoutError(failed)
```

Comment:
This will make us error out after checking out as much as we can (and before we save the dir entry in state).

This is a change from the current behavior, and I'm not sure how exactly we want to handle partial checkouts. (Should it just be a warning instead of an error?)